.github/generate-quest-list.main.kts

```diff
@@ -0,0 +1,204 @@
+/**
+ * Note: Run from the project's root directory like this:
+ * kotlinc -script -Xplugin=/path/to/kotlinx-serialization-compiler-plugin.jar .github/generate-quest-list.main.kts
+ */
+
+@file:DependsOn("org.jetbrains.kotlinx:kotlinx-serialization-core:1.0.1")
+@file:DependsOn("org.jetbrains.kotlinx:kotlinx-serialization-json:1.0.1")
+@file:Suppress("PLUGIN_IS_NOT_ENABLED")
+
+import java.io.File
+import java.net.URL
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+val projectDirectory = File(".")
+val sourceDirectory = projectDirectory.resolve("app/src/main/java/de/westnordost/streetcomplete/")
+
+val csvFile = projectDirectory.resolve("quest-list.csv")
+
+val noteQuestName = "OsmNoteQuest"
+val noteQuestFile = sourceDirectory.resolve("data/osmnotes/notequests/OsmNoteQuestType.kt")
+
+val wikiRowSpan2 = " rowspan=\"2\" |"
+
+val questFileContent = sourceDirectory.resolve("quests/QuestModule.kt").readText()
+val questNameRegex = Regex("(?<=^ {8})[A-Z][a-zA-Z]+(?=\\()", RegexOption.MULTILINE)
```

Comment:
Ui, I suppose this is ok, though this might be a little unstable.

Reply:
It is indeed. Can you think of a smarter way?

Reply:
Maybe look at all files in the `/quests` directory and search for `class ([A-Z][a-zA-Z]+)\s*:\s*(OsmElementQuestType|OsmFilterQuestType|QuestType)`, i.e. all classes that inherit from QuestType or a subclass/interface of it.
This is also a bit unstable of course, because there might be new subclasses/interfaces in the future.

Reply:
We'd still need to parse the `QuestModule` though, to find out the order (default priority).

Reply:
I think it's fine if the regexes are unstable — as long as updating the script is less work than manually updating the wiki, it's a net gain.

That said, it may not be a net gain if the regex breaks *silently* and produces wrong results. I've only read the comments here, not dug into the code, but I wonder if there is a way to put a sanity check on the regex result and have it fail if the sanity check fails. I'm thinking something along the lines of a hard-coded assertion about the number of quest names we expect to be found, where the script fails if it's the wrong number. Of course, this will break every time a quest is added or removed, but that's just the point. If it's expected breakage (because a quest was added/removed), it's simple enough to update the number. If it's unexpected, then the check has done its job.

Reply:
right

Reply:
Sorry I meant to comment sooner. Now this is also Kotlin can't you just load QuestModule and access the data from that directly the same way the app does? Which will then be fine until QuestModule gets changed.

Reply:
That seems not possible, because the script is not built together with the other source files. Also if it was, we'd need to supply suitable (probably mocked) parameters.

If someone finds a solution for this, feel free to change it :)